pkg/block/block.go

```diff
@@ -95,3 +101,59 @@ func renameFile(from, to string) error {
 	}
 	return pdir.Close()
 }
+
+// DownloadDir downloads directory that is mean to be block directory.
+func DownloadDir(ctx context.Context, bucket objstore.Bucket, id ulid.ULID, dst string) error {
+	if err := objstore.DownloadDir(ctx, bucket, id.String(), dst); err != nil {
+		return err
+	}
+
+	chunksDir := filepath.Join(dst, "chunks")
+	_, err := os.Stat(chunksDir)
+	if os.IsNotExist(err) {
+		// This can happen if block is empty. We cannot easily upload empty directory, so create one here.
+		return os.Mkdir(chunksDir, os.ModePerm)
+	}
+
+	if err != nil {
+		return errors.Wrapf(err, "stat %s", chunksDir)
+	}
+
+	return nil
+}
+
+// DeleteDir removes directory that is mean to be block directory.
+// NOTE: Prefer this method instead of objstore.DeleteDir to avoid deleting empty dir (whole bucket) by mistake.
+func DeleteDir(ctx context.Context, bucket objstore.Bucket, id ulid.ULID) error {
```

Comment:
Why not `DeleteBlock` and `DownloadBlock` then?

Reply:
because we are in `block` package... so maybe just `Download` and `Delete`?

Reply:
Fine with me as well. Just `Dir` we bugging me since the whole point of these helpers is that we don't just hit an arbitrary dir.